junit-platform-launcher/src/main/java/org/junit/platform/launcher/listeners/SummaryGeneratingListener.java

```diff
@@ -123,10 +122,6 @@ public void executionFinished(TestIdentifier testIdentifier, TestExecutionResult
 					throwable -> this.summary.addFailure(testIdentifier, throwable));
 				break;
 			}
-
-			default:
-				throw new PreconditionViolationException(
-					"Unsupported execution status:" + testExecutionResult.getStatus());
```

Comment:
I'm not a fan of that proposal either.

As I stated elsewhere, I always either include all enum constants in a switch statement or a `default` block that logs a warning or throws an exception.

Moving the _default_ behavior outside of the switch statement is something I would like to avoid.